src/jit/legacyjit/CMakeLists.txt

```diff
@@ -18,6 +18,7 @@ endif(WIN32)
 add_library_clr(legacyjit
    SHARED
    ${SHARED_LIB_SOURCES}
+   ${JIT_ARCH_SOURCES}
```

Comment:
The legacy sources are included in all builds; it's the `LEGACY_BACKEND` definition here (and not elsewhere) that causes them to actually be used.

This is by design, for simplicity, and to improve IntelliSense.